src/Compilers/CSharp/Portable/Binder/Binder_Constraints.cs

```diff
@@ -383,7 +380,7 @@ private TypeParameterConstraintClause GetDefaultTypeParameterConstraintClause(Ty
                     // since, in general, it may be difficult to support all invalid types.
                     // In the future, we may want to include some invalid types
                     // though so the public binding API has the most information.
-                    if (IsValidConstraint(typeParameter.Name, syntax, constraintType, constraintClause.Constraints, constraintTypeBuilder, diagnostics))
+                    if (IsValidConstraint(typeParameter, syntax, constraintType, constraintClause.Constraints, constraintTypeBuilder, performOnlyCycleSafeValidation, diagnostics))
```

Comment:
>performOnlyCycleSafeValidation [](start = 134, length = 30)

Might we end up removing different constraints in the two binding passes (two values of performOnlyCycleSafeValidation)?

Reply:
> Might we end up removing different constraints in the two binding passes (two values of performOnlyCycleSafeValidation)?

Yes. ```performOnlyCycleSafeValidation == false``` can remove more types than ```performOnlyCycleSafeValidation == true```

---
In reply to: [520070230](https://github.com/dotnet/roslyn/pull/49227#discussion_r520070230) [](ancestors = 520070230)